src/DnsHelper.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Yiisoft\NetworkUtilities;
+
+class DnsHelper
+{
+    /**
+     * @param string $hostname hostname without dot at end
+     */
+    public static function checkMx(string $hostname): bool
+    {
+        $hostname .= '.';
+        try {
+            if (!@dns_check_record($hostname, 'MX')) {
+                return false;
+            }
+            $result = @dns_get_record($hostname, DNS_MX);
+            return $result !== false && count($result) > 0;
+        } catch (\Throwable $t) {
+        }
+        return false;
+    }
+
+    public static function checkA(string $hostname): bool
+    {
+        try {
+            if (!@dns_check_record($hostname, 'A')) {
+                return false;
+            }
+            $result = @dns_get_record($hostname, DNS_A);
+            return $result !== false && count($result) > 0;
+        } catch (\Throwable $t) {
+        }
+        return false;
+    }
+
+    public static function checkForEmail(string $hostnameOrEmail): bool
```

Comment:
I'd like to choose cleaner name like `isAcceptingServerForEmail`. @cebe, @SilverFire any suggestions?

Reply:
`DnsHelper::isAcceptingServerForEmail()` makes me think about accepting of the DNS server. I'd not use the `server` word in this context.

Maybe `DnsHelper::domainAcceptsEmails()`?

Reply:
> Maybe DnsHelper::domainAcceptsEmails()?

Like. Let me change it a bit: `DnsHelper::acceptsEmails($hostname)` or `DnsHelper::isAccpetsEmails()`. The whole helper is all about DNS, so I think the `domain` prefix is ​​just an overdetermination of the function.

Reply:
`acceptsEmails()` sounds great